lib/nats.js

```diff
@@ -742,14 +764,21 @@ Client.prototype.setupHandlers = function () {
     }
 
     // Only bubble up error if we never had connected
-    // to the server and we only have one.
+    // to the server and we only have one, and close
     if (this.wasConnected === false && this.servers.length === 0) {
       this.emit('error', new NatsError(CONN_ERR_MSG_PREFIX + exception, CONN_ERR, exception))
+      this.close()
+      return
     }
+    // continue with reconnect
     this.closeStream()
   })
 
   stream.on('data', (data) => {
+    if (this.options.timeout) {
+      // connect added a timeout on data from the socket - remove it
+      stream.setTimeout(0)
```

Comment:
This is going to call this for every incoming message, does not seem efficient.

Reply:
zonk!